mewarpx/mewarpx/emission.py

```diff
@@ -1179,20 +1180,155 @@ def get_normals(self, x, y, z):
         return normals
 
 
+class XPlaneEmitter(Emitter):
+    """Injection for a planar cathode emitting from the simulation side."""
+
+    geoms = ['XZ', 'XYZ']
+
+    def __init__(self, conductor, T=None, x=None, ymin=None, ymax=None,
+                 zmin=None, zmax=None, transverse_fac=1.0, xsign=1, **kwargs):
+        """Initialize an emitter for a planar cathode.
+
+        Arguments:
+            conductor (:class:`mewarpx.assemblies.Assembly`): Conductor object,
+                used to obtain work function and z coordinate/direction.
+            T (float): Temperature in Kelvin for the emitter; determines
+                velocities. If not specified the temperature of the conductor
+                will be used.
+            x (float): Position of the emitter along the x axis. Default
+                conductor.x if it exists otherwise conductor.xmin/max depending
+                on xsign. If none of those attributes exist an error will be
+                raised if x is not specified.
+            ymin (float): Minimum position of the rectangular emitter along y.
+                Default conductor.ymin if it exists otherwise mwxrun.ymin.
+            ymax (float): Maximum position of the rectangular emitter along y.
+                Default conductor.ymax if it exists otherwise mwxrun.ymax.
+            zmin (float): Minimum position of the rectangular emitter along z.
+                Default conductor.zmin if it exists otherwise mwxrun.zmin.
+            zmax (float): Maximum position of the rectangular emitter along z.
+                Default conductor.zmax if it exists otherwise mwxrun.zmax.
+            transverse_fac (float): Scale the transverse energy distribution by
+                this factor. Default 1. See
+                :func:`mewarpx.utils_store.util.get_velocities` for details.
+            xsign (int): 1 to emit in +x, -1 to emit in -x.
+            kwargs (dict): Any other keyword arguments supported by the parent
+                Emitter constructor (such as "emission_type").
+        """
+        # Default initialization
+        super(XPlaneEmitter, self).__init__(T=T, conductor=conductor, **kwargs)
+
+        self.transverse_fac = transverse_fac
+        self.xsign = int(round(xsign))
+        if self.xsign not in [-1, 1]:
+            raise ValueError("xsign must be +1 or -1 for x-plane emitters.")
+
+        self.x = x
+        if self.x is None:
+            try:
+                self.x = conductor.x
+            except AttributeError:
+                try:
+                    attr = 'xmax' if self.xsign else 'xmin'
```

Comment:
`bool(1) == bool(-1) == True`, so I don't think this logic works?

Reply:
Thanks!